test_shopcart.py

```diff
@@ -29,6 +29,175 @@ def test_index(self):
         self.assertTrue ('Shopcart Demo REST API Service' in resp.data)
 
 
+    def test_create_shopcarts(self):
```

Comment:
I think the test functions are too long. Each test should be simple, easy to read. 
It seems to me that each block of your code should be one test function, and the comments should be used as test function names.